core/templates/dev/head/pages/exploration_editor/editor_tab/StateInteraction.js

```diff
@@ -337,17 +343,20 @@ oppia.controller('StateInteraction', [
             $scope.reallyDelete = function() {
               $uibModalInstance.close();
             };
-
             $scope.cancel = function() {
               $uibModalInstance.dismiss('cancel');
               AlertsService.clearWarnings();
             };
           }
         ]
       }).result.then(function() {
+        // TODO(SD): Clean Hints and Responses to resolve #4634.
```

Comment:
@seanlip, is this an expected behavior?
We don't clean the stateHintsService while deleting an interaction? (I'll have to do the same thing for the audio part)

Reply:
(Hm, looks like my comment showed up in the main thread. I'll reply inline.)

Yep I think it is expected. Scenario: creator wants to ask exactly the same question, but using a different interaction (eg number with units instead of just a number).

Reply:
@seanlip,
- Create an exploration add textInput interaction.
- Add hint.
- Delete Interaction
- Add end exploration interaction.
- Preview
- Hint appears in "End exploration" card.

Is this expected?

Reply:
Oh, that's a good point. I didn't think of that scenario. It makes sense for hints not to appear when the state does not have an interaction to which answers are submittable in the first place.

Could you please file a bug for this? I am open to fixing it in this PR or fixing it in a separate PR, and I think the latter might be better since this PR is getting large, but your call.

Reply:
Sure! (I have created an issue for now #5063 I'll fix this in another PR)